lxd/storage/drivers/driver_zfs.go

```diff
@@ -560,6 +560,30 @@ func (d *zfs) importPool() (bool, error) {
 	return false, fmt.Errorf("ZFS zpool exists but dataset is missing")
 }
 
+func (d *zfs) isSetup() (bool, error) {
+	setup, err := d.datasetExists(d.config["zfs.pool_name"])
+	if err != nil {
+		return false, err
+	}
+
+	if !setup {
```

Comment:
Just a nit, to safe this extra condition here and below you can just do `if err != nil || !setup` when checking for existence of the dataset as in the latter case `err` equals `nil` anyway.